COAP server and client not working with POSIX

When compiling the POSIX version of OpenThread the `coap server` and `coap client` commands do not work.

The commands do not show up with the `help` command either.

I have the latest version of the source code downloaded, and I am building it using:
```
make -f examples/Makefile-posix
```
Am I missing a necessary flag to get this to compile properly, or are the COAP commands just incompatible with the POSIX version?

I tried that earlier and I got the following error:
```
Undefined symbols for architecture x86_64:
  "Thread::Cli::Coap::Process(otInstance*, int, char**, Thread::Cli::Server&)", referenced from:
      Thread::Cli::Interpreter::ProcessCoap(int, char**) in libopenthread-cli.a(libopenthread_cli_a-cli.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[5]: *** [ot-cli-mtd] Error 1
make[5]: *** Waiting for unfinished jobs....
Undefined symbols for architecture x86_64:
  "Thread::Cli::Coap::Process(otInstance*, int, char**, Thread::Cli::Server&)", referenced from:
      Thread::Cli::Interpreter::ProcessCoap(int, char**) in libopenthread-cli.a(libopenthread_cli_a-cli.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```
However using a different machine, where the architecture is `i686` it works without any problems.

Perhaps the issue is specific to some architectures

That seemed to help, but it is now getting a different error
```
/Users/Username/Documents/Git/openthread/examples/../src/cli/cli_coap.cpp:83:22: error: implicit
      conversion loses integer precision: 'int' to 'char' [-Werror,-Wconversion]
        aString[i] = tolower(aString[i]);
                   ~ ^~~~~~~~~~~~~~~~~~~
1 error generated.
```

--- a/.travis/script.sh
+++ b/.travis/script.sh
@@ -37,7 +37,7 @@ set -x
 [ $BUILD_TARGET != pretty-check ] || {
     export PATH=/tmp/astyle/build/gcc/bin:$PATH || die
     ./bootstrap || die
-    ./configure --enable-ftd --enable-cli --enable-diag --enable-dhcp6-client --enable-dhcp6-server --enable-dns-client --enable-commissioner --enable-joiner --with-examples=posix || die
+    ./configure --enable-ftd --enable-cli --enable-diag --enable-dhcp6-client --enable-dhcp6-server --enable-dns-client --enable-commissioner --enable-joiner --enable-application-coap --with-examples=posix || die
     make pretty-check || die
 }
 

--- a/src/cli/cli_coap.cpp
+++ b/src/cli/cli_coap.cpp
@@ -72,15 +72,17 @@ void Coap::PrintPayload(otMessage *aMessage)
             bytesPrinted += bytesToPrint;
         }
     }
+
     sServer->OutputFormat("\r\n");
 }
 
 void Coap::ConvertToLower(char *aString)
 {
     uint8_t i = 0;
+
     while (aString[i])
     {
-        aString[i] = tolower(aString[i]);
+        aString[i] = (char)tolower(aString[i]);
         i++;
     }
 }
@@ -104,6 +106,7 @@ ThreadError Coap::Process(otInstance *aInstance, int argc, char *argv[], Server 
     sResource.mHandler = (otCoapRequestHandler) &Coap::s_HandleServerResponse;
 
     VerifyOrExit(argc > 0, error = kThreadError_InvalidArgs);
+
     for (unsigned int i = 0; i < sizeof(sCommands) / sizeof(sCommands[0]); i++)
     {
         if (strcmp(argv[0], sCommands[i].mName) == 0)
@@ -124,6 +127,7 @@ ThreadError Coap::ProcessServer(int argc, char *argv[])
     VerifyOrExit(argc > 0, error = kThreadError_InvalidArgs);
 
     ConvertToLower(argv[0]);
+
     if (strcmp(argv[0], "start") == 0)
     {
         SuccessOrExit(error = otCoapServerStart(sInstance));
@@ -270,6 +274,7 @@ ThreadError Coap::ProcessClient(int argc, char *argv[])
 
     // CoAP-Code
     ConvertToLower(argv[0]);
+
     if (strcmp(argv[0], "get") == 0)
     {
         coapCode = kCoapRequestGet;
@@ -313,6 +318,7 @@ ThreadError Coap::ProcessClient(int argc, char *argv[])
 
     // CoAP-Type
     ConvertToLower(argv[3]);
+
     if ((argc > 3) && (strcmp(argv[3], "con") == 0))
     {
         coapType = kCoapTypeConfirmable;
@@ -339,21 +345,23 @@ ThreadError Coap::ProcessClient(int argc, char *argv[])
 
     if ((coapType == kCoapTypeConfirmable) || coapCode == kCoapRequestGet)
     {
-    	error = otCoapSendRequest(sInstance, message, &messageInfo,
-    	                              (otCoapResponseHandler) &Coap::s_HandleClientResponse, NULL);
+        error = otCoapSendRequest(sInstance, message, &messageInfo,
+                                  (otCoapResponseHandler) &Coap::s_HandleClientResponse, NULL);
     }
     else
     {
-    	error = otCoapSendRequest(sInstance, message, &messageInfo, NULL, NULL);
+        error = otCoapSendRequest(sInstance, message, &messageInfo, NULL, NULL);
     }
 
     sServer->OutputFormat("Sending CoAP message: ");
 
 exit:
+
     if ((error != kThreadError_None) && (message != NULL))
     {
         otMessageFree(message);
     }
+
     return error;
 }
 

